src/theory/datatypes/inference_manager.cpp

```diff
@@ -65,6 +76,34 @@ void InferenceManager::process()
   doPendingFacts();
 }
 
+void InferenceManager::sendDtLemma(Node lem,
+                                   InferId id,
+                                   LemmaProperty p,
+                                   bool doCache)
+{
+  if (isProofEnabled())
+  {
+    processDtLemma(lem, Node::null(), id);
```

Comment:
Just for readability (and deal with stats in one place):
```suggestion
    if (processDtLemma(lem, Node::null(), id))
    {
      d_inferenceLemmas << id;
    }
```

Reply:
Actually, this would not keep stats if proofs were disabled, right?

Reply:
it would. If proofs are disabled, then stats are collected immediately below.

Reply:
So I see the issue and now remember why I did it like this.  These are two different methods, this one sends lemmas immediately, the other one processed lemmas that were buffered.  The two control paths are disjoint when proofs are disabled.  This change would make it so that lemmas that were buffered and processed would not be counted when proofs are disabled.